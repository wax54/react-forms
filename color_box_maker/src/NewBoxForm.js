import React, { useState } from "react";


const NewBoxForm = ({onSubmit}) => {
    const INITIAL_FORM_DATA = {
        height: '',
        width: '',
        color: ''
    };
    const [formData, setFormData] = useState(INITIAL_FORM_DATA); 

    const handleChange = evt => {
        const { name, value } = evt.target;
        setFormData(data => ({...data, [name]: value}))
    }
    const handleSubmit = evt => {
        evt.preventDefault();
        onSubmit(formData);
        setFormData(INITIAL_FORM_DATA);
    }

    return (
    <form onSubmit={handleSubmit}>
        <label htmlFor="height-input">Height: </label>
        <input id="height-input" 
            type="number" 
            name="height" 
            min="0"
            value={formData.height} 
            onChange={handleChange}
            required
        />
        <label htmlFor="width-input">Width: </label>
        <input id="width-input" 
            type="number" 
            name="width" 
            min="0"
            value={formData.width} 
            onChange={handleChange}
            required
        />
        
        <label htmlFor="color-input">Color: </label>
        <input id="color-input" 
            type="text" 
            name="color" 
            value={formData.color}
            onChange={handleChange}
            required
        />
        <button>Make me a Box!</button>
    </form>
    )
};

export default NewBoxForm;
