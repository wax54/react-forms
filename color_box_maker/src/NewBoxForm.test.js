import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import NewBoxForm from './NewBoxForm';

it("Doesn't Blow up", () => {
    render(<NewBoxForm />);
});

it('should match previous snapshot', function () {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});

it("should allow users to type into the inputs", () =>{
    const { getByLabelText, getByText } = render(<NewBoxForm />);

    const heightInput = getByLabelText("Height: ");
    const widthInput = getByLabelText("Width: ");
    const colorInput = getByLabelText("Color: ");

    const btn = getByText("X");


    fireEvent.change(heightInput, { target: { value: "100" } });
    fireEvent.change(widthInput, { target: { value: "200" } });
    fireEvent.change(colorInput, { target: { value: "blue" } });

    fireEvent.click(btn);

    expect()
})


