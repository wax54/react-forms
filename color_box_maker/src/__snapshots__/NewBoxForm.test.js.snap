// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match previous snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  />
  <button>
    X
  </button>
</DocumentFragment>
`;
